docs/2-fork-accountability.md

```diff
@@ -0,0 +1,163 @@
+
+By receiving messages logs from correct processes, the monitor would be able to determine if some correct process received more than one message with the same type from another process p (i.e., if some process equivocated).  
+However, we have also shown that a process can be faulty in other ways and that would not be enough for designing an asynchronous accountability algorithm that catches all the faulty processes that led to a fork. 
+In fact, it can happen that, by not receiving message logs from one message, we might end up mistakenly detecting correct processes because we do not have enough information to determine if a process has enough justifications to have sent a Prevote/Precommit message (accuracy property violated). 
+
+If we want to design a correct accountability algorithm, we need to slightly modify the Tendermint consensus algorithm in order to correctly respect the accuracy property.
+As we said above, we need to catch the case where a fork happens without having any process equivocating.
+
+By looking at the other faultiness reasons, we can see that the critical case is when a process sends a Precommit message for a value v and then, in a later round, it sends a Prevote message for a different value than v.
+The asynchronous accountability algorithm would work correctly if processes can justify the sending of a Prevote message: a simple solution is to attach the justifications to the Prevote message itself so that the accountability algorithm can directly check if the message has been correctly sent.
+
+This would be enough because, intuitively, there will be at least one correct process that has received both messages and it would be possible to track down the origin and the justifications and find out if a process is really faulty.
+
+Therefore, the monitor will be able to complete the accountability algorithm as soon as at least f+1 processes will be detected.
+We also know that at least f+1 messages logs will be received by the monitor (we assume correct processes will send their logs). The algorithm will be able to complete the algorithm and respect the completeness and accuracy properties with the message logs of the correct processes.
+
+## Algorithm design
+ 
+To summarize the above discussion, these are the high-level steps carried out by the monitor to run the asynchronous accountability algorithm:
+
+1. Upon a fork, monitor requests message logs from all the validators
+
+2. Monitor runs the accountability algorithm to detect faulty processes upon receiving at least *f + 1* message logs and at every new message logs received when there are at least *f + 1* message logs 
+
+3. While running the accountability algorithm, the monitor analyzes the received logs and build the correct logs for each process: for every message *m* that has been received by some other process but is not present in the sender's message logs, the monitor attaches *m* to the message logs of the sender of *m*. 
```

Comment:
It is not clear what a "correct log" is.
But, this is nitpicking. We will discuss today the level of details we want to use here.

Reply:
Fixed